### PR TITLE
Task/hw 53712 add accessors to bank account bank card paypal account

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift [hunk 1]
```diff
@@ -53,32 +53,32 @@ class HyperwalletTransferMethodTests: XCTestCase {
...
-        XCTAssertEqual(bankCard?.getField(fieldName: .type) as? String, "BANK_CARD")
-        XCTAssertEqual(bankCard?.getField(fieldName: .token) as? String, "trm-00002")
+        let bankCard = transferMethods?.data.first { ($0.type ?? "")  == "BANK_CARD" }
+
+        XCTAssertEqual(bankCard?.type, "BANK_CARD")
+        XCTAssertEqual(bankCard?.token, "trm-00002")
+        XCTAssertEqual(bankCard?.status, "ACTIVATED")
+        XCTAssertEqual(bankCard?.transferMethodCountry, "US")
+        XCTAssertEqual(bankCard?.transferMethodCurrency, "USD")
         XCTAssertEqual(bankCard?.getField(fieldName: .cardNumber) as? String, "************1358")
```
Comment (oleksii-lubianyi-epam) [inline Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: @jsaini-hw 
As `transferMethods` is of type `HyperwalletPageList<HyperwalletTransferMethod>?`, only instances of base class `HyperwalletTransferMethod` instantiated and attempts to cast down fail.
Leaving as it is for now, will appreciate for a piece of advice.

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift [hunk 2]
```diff
@@ -53,32 +53,32 @@ class HyperwalletTransferMethodTests: XCTestCase {
...
-        XCTAssertEqual(bankAccount?.getField(fieldName: .token) as? String, "trm-00001")
+        let bankAccount = transferMethods?.data.first { ($0.type ?? "")  == "BANK_ACCOUNT" }
+
+        XCTAssertEqual(bankAccount?.type, "BANK_ACCOUNT")
+        XCTAssertEqual(bankAccount?.token, "trm-00001")
+        XCTAssertEqual(bankAccount?.status, "ACTIVATED")
+        XCTAssertEqual(bankAccount?.createdOn, "2018-12-15T00:30:12")
+        XCTAssertEqual(bankAccount?.transferMethodCountry, "US")
+        XCTAssertEqual(bankAccount?.transferMethodCurrency, "USD")
         XCTAssertEqual(bankAccount?.getField(fieldName: .bankAccountId) as? String, "25589087")
```
Comment (oleksii-lubianyi-epam) [inline Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: @jsaini-hw 
As `transferMethods` is of type `HyperwalletPageList<HyperwalletTransferMethod>?`, only instances of base class `HyperwalletTransferMethod` instantiated and attempts to cast down fail.
Leaving as it is for now, will appreciate for a piece of advice.

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift (hunk 1 above)
Comment (jsaini-hw) [reply Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: Modify `public func getField(fieldName: TransferMethodField) -> String? {
        return self.storage[fieldName.rawValue]?.value as? String
    }`
Fix all the warnings and leave it as it is that can not be changed to use getters.

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift (hunk 2 above)
Comment (jsaini-hw) [reply Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: Same as above

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift (hunk 1 above)
Comment (oleksii-lubianyi-epam) [reply Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: Thanks, fixed

File: Tests/TransferMethod/HyperwalletTransferMethodTests.swift (hunk 2 above)
Comment (oleksii-lubianyi-epam) [reply Tests/TransferMethod/HyperwalletTransferMethodTests.swift]: Thanks, fixed

**Comment by jsaini-hw** [comment]
@oleksii-lubianyi-epam code coverage in master is 98% and for this PR it's showing 94%. Please check where code coverage decreased and try to add more tests to make it 98%

**Pull request merged**